haystack/nodes/ranker/recentness_ranker.py

```diff
@@ -0,0 +1,147 @@
+                Continuing without sorting by date.
+                """,
+                self.date_identifier,
+                ",".join(list(documents[0].meta.keys())),
+            )
+            output = {"documents": documents}
+            return output, "output_1"
+        except ParserError:
+            logger.error(
+                """
+                Could not parse date information for dates: %s\n
+                Continuing without sorting by date.
+                """,
+                " - ".join([x.meta.get(self.date_identifier, "identifier wrong") for x in documents]),
+            )
+            output = {"documents": documents}
+            return output, "output_1"
+
+        # merge scores for both documents sorted by content and date.
+        # We use reciprocal rank fusion (RRF) to have the same range of scores for both lists.
+        scores_map: Dict = defaultdict(int)
+        document_map = {doc.id: doc for doc in documents}
+        weight = self.weight
+        for i, doc in enumerate(documents):
+            if self.method == "reciprocal_rank_fusion":
+                scores_map[doc.id] += self._calculate_rrf(rank=i) * (1 - weight)
+            elif self.method == "score":
+                score = float(0)
+                if doc.score is None:
+                    warnings.warn("The score was not provided; defaulting to 0")
+                elif doc.score < 0 or doc.score > 1:
+                    warnings.warn("The score is outside the [0,1] range; defaulting to 0")
+                else:
+                    score = doc.score
+
+                scores_map[doc.id] += score * (1 - weight)
+            else:
+                logger.error(
+                    """
+                    Param <method> seems wrong.\n
+                    You supplied: '%s'.\n
+                    It should be 'reciprocal_rank_fusion' or 'score'
+                    """,
+                    self.method,
+                )
+        for i, doc in enumerate(sorted_by_date):
+            if self.method == "reciprocal_rank_fusion":
+                scores_map[doc.id] += self._calculate_rrf(rank=i) * weight
+            elif self.method == "score":
+                scores_map[doc.id] += self._calc_recentness_score(rank=i, amount=len(sorted_by_date)) * weight
+        sorted_doc_ids = sorted(scores_map.items(), key=lambda d: d[1] if d[1] is not None else -1, reverse=True)
+
+        top_k = top_k or self.top_k or len(sorted_doc_ids)
+        docs = []
+        for idx, score in sorted_doc_ids[:top_k]:
+            doc = document_map[idx]
+            doc.score = score
+            docs.append(doc)
+
+        output = {"documents": docs}
+
+        return output, "output_1"
+
+    # pylint: disable=arguments-differ
+    def run_batch(self, queries: List[str], documents: Union[List[Document], List[List[Document]]], top_k: Optional[int] = None, batch_size: Optional[int] = None) -> Tuple[Dict, str]:  # type: ignore
```

Comment:
I think this line's a bit long and hard to read. Might be good to line-separate the arguments, though that may just be a matter of taste. (The definition for `run` is shorter, though perhaps you might want to do the same there as well.)

Reply:
I've modified this one so that all parameters are on a separate line, but for the "run" (which is now "predict") the pre-commit hooks didn't allow me to do so - every time I made the parameters line-separated they would automatically go back to single line before committing. I guess it only allows this line separation from 5+ parameters or maybe from a specific length in characters...